pkg/handlers/syncResources.go

```diff
@@ -82,6 +82,11 @@ func SyncResources(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, "Aggregator has many pending requests, retry later.", http.StatusTooManyRequests)
 		return
 	}
+	if _, exist := PendingRequests[clusterName]; exist {
```

Comment:
@jlpadilla Should this block of code run before the previous block which checks the count. If a second request comes and we already have max in process, we want to say you are already processing than saying `too many` .

Reply:
done